### PR TITLE
EkncDomain: support multiple subscriptions

### PR DESCRIPTION
**Pull request by xjuan**
https://phabricator.endlessm.com/T17999

### CODE REVIEW

File: ekncontent/ekncontent/eknc-domain.c [hunk 1]
```diff
@@ -416,7 +368,38 @@ eknc_domain_initable_init (GInitable *initable,
...
+    g_string_append (root_from_tmp, "../");
+
+  for (l = dbs; l; l = g_list_next (l))
+    {
+      JsonObject *json_db = json_node_get_object (l->data);
+
+      guint64 offset = json_object_get_int_member (json_db, "offset");
+      const char *path = json_object_get_string_member (json_db, "path");
+
+      g_output_stream_printf (stream, NULL, NULL, NULL,
```
Comment (tchx84) [inline ekncontent/ekncontent/eknc-domain.c]: question: wouldn't be better to create this structure and serialize it?

File: ekncontent/ekncontent/eknc-domain.c [hunk 2]
```diff
@@ -445,34 +428,104 @@ eknc_domain_initable_init (GInitable *initable,
...
+  EkncDomain *self = EKNC_DOMAIN (initable);
+
+  gboolean has_app_id = (self->app_id != NULL && *self->app_id != '\0');
+  gboolean has_path = (self->path != NULL && *self->path != '\0');
+
+  if (has_path)
+    {
+      g_autoptr(GFile) subscription_dir = g_file_new_for_path (self->path);
+      self->manifest_file = g_file_get_child (subscription_dir, "manifest.json");
+
```
Comment (tchx84) [inline ekncontent/ekncontent/eknc-domain.c]: nit: white space.

File: ekncontent/ekncontent/eknc-domain.c [hunk 3]
```diff
@@ -445,34 +428,104 @@ eknc_domain_initable_init (GInitable *initable,
...
+      bundle_subscriptions_dir = g_file_get_child (self->content_dir, "com.endlessm.subscriptions");
+
+      GList *l;
+      for (l = self->subscriptions; l; l = g_list_next (l))
+        {
+          const gchar *subscription_id = l->data;
+          g_autoptr(GFile) subscription_dir = g_file_get_child (subscriptions_dir, subscription_id);
+          g_autoptr(GFile) bundle_dir = g_file_get_child (bundle_subscriptions_dir, subscription_id);
+
+          eknc_domain_process_subscription (self,
```
Comment (tchx84) [inline ekncontent/ekncontent/eknc-domain.c]: nit: trailing space.

**Comment by tchx84** [comment]
Another thing I noticed is that when using a custom shard path via `--content-path`, I get a lot of these 
errors:
```
(com.endlessm.animals.en:20080): GLib-GIO-CRITICAL **: g_output_stream_vprintf: assertion 'G_IS_OUTPUT_STREAM (stream)' failed

```

File: ekncontent/ekncontent/eknc-domain.c (hunk 1 above)
Comment (xjuan) [inline ekncontent/ekncontent/eknc-domain.c]: Yes, but not much better, it still an ugly hack!

**Pull request merged**